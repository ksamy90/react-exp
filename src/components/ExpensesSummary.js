import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({ expenseCount, amount }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const expensesTotal = numeral(amount / 100).format('$0,0.00');
    if (expenseCount === 0) {
        return <p></p>;
    };

    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {expensesTotal}</h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        amount: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
