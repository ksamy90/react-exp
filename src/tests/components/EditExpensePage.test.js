import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let expense, removeExpense, editExpense, history, wrapper;
beforeEach(() => {
    expense = expenses[2];
    editExpense = jest.fn();
    history = { push: jest.fn() };
    removeExpense = jest.fn();
    wrapper = shallow(<EditExpensePage 
        expense={expense} 
        editExpense={editExpense} 
        history={history} 
        removeExpense={removeExpense} 
    />);
});

test('should render EditExpensepage ', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle editExpense', () => {
    wrapper.find('ExpenseForm').prop('editExpense')(expense);
    expect(editExpense).toHaveBeenLastCalledWith(expense.id, expense);
    expect(history.push).toHaveBeenLastCalledWith('/');
});

test('should handle removeExpense', () => {
    wrapper.find('button').simulate('click');
    expect(removeExpense).toHaveBeenLastCalledWith({ id: expense.id });
    expect(history.push).toHaveBeenLastCalledWith('/');
});
