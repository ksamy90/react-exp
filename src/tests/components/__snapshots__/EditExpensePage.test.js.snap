// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensepage  1`] = `
<div>
  <ExpenseForm
    editExpense={[Function]}
    expense={
      Object {
        "amount": 4500,
        "createdAt": 345600000,
        "description": "Credit Card",
        "id": "3",
        "note": "",
      }
    }
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
